pkg/storageos/daemonset.go

```diff
@@ -0,0 +1,262 @@
+					HostPID:            true,
+					HostNetwork:        true,
+					DNSPolicy:          corev1.DNSClusterFirstWithHostNet,
+					InitContainers: []corev1.Container{
+						{
+							Name:  "enable-lio",
+							Image: s.stos.Spec.GetInitContainerImage(),
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "kernel-modules",
+									MountPath: "/lib/modules",
+									ReadOnly:  true,
+								},
+								{
+									Name:             "sys",
+									MountPath:        "/sys",
+									MountPropagation: &mountPropagationBidirectional,
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: &privileged,
+								Capabilities: &corev1.Capabilities{
+									Add: []corev1.Capability{"SYS_ADMIN"},
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Image: s.stos.Spec.GetNodeContainerImage(),
+							Name:  "storageos",
+							Args:  []string{"server"},
+							Ports: []corev1.ContainerPort{{
+								ContainerPort: 5705,
+								Name:          "api",
+							}},
+							LivenessProbe: &corev1.Probe{
+								InitialDelaySeconds: int32(65),
+								TimeoutSeconds:      int32(10),
+								FailureThreshold:    int32(5),
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/v1/health",
+										Port: intstr.IntOrString{Type: intstr.String, StrVal: "api"},
+									},
+								},
+							},
+							ReadinessProbe: &corev1.Probe{
+								InitialDelaySeconds: int32(65),
+								TimeoutSeconds:      int32(10),
+								FailureThreshold:    int32(5),
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/v1/health",
+										Port: intstr.IntOrString{Type: intstr.String, StrVal: "api"},
+									},
+								},
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name: hostnameEnvVar,
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "spec.nodeName",
+										},
+									},
+								},
+								{
+									Name: adminUsernameEnvVar,
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: initSecretName,
+											},
+											Key: "username",
+										},
+									},
+								},
+								{
+									Name: adminPasswordEnvVar,
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: initSecretName,
+											},
+											Key: "password",
+										},
+									},
+								},
+								{
+									Name:  joinEnvVar,
+									Value: s.stos.Spec.Join,
+									// ValueFrom: &corev1.EnvVarSource{
+									// 	FieldRef: &corev1.ObjectFieldSelector{
+									// 		FieldPath: "status.podIP",
+									// 	},
+									// },
```

Comment:
Remove?

Reply:
Yes, we should remove that. We will not need that again.